File: Java/DesignPatterns/src/test/java/designPatterns/structural/facade/MainFacadeTest.java
package designPatterns.structural.facade;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainFacadeTest {
  private Facade facade;

  @BeforeEach
  void setUp() {
    facade = new Facade();
  }

  @Test
  void equalsTest() {
    assertThat(facade.start()).isEqualTo("SubClassA, 100, true");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/structural/facade/MainFacadeTest.kt
package structural.facade

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainFacadeTest {
  private lateinit var facade: Facade

  @BeforeEach
  internal fun setUp() {
    facade = Facade()
  }

  @Test
  internal fun equalsTest() {
    assertThat(facade.start()).isEqualTo("SubClassA, 100, true")
  }
}